File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.swift
class Solution {
	func divideString(_ s: String, _ k: Int, _ fill: Character) -> [String] {
		var result: [String] = []
		var i = 0
		
		while i < s.count {
			let start = s.index(s.startIndex, offsetBy: i)
			var end = s.index(s.startIndex, offsetBy: s.count - 1)
			
			// print(i + k - 1)
			// print(s.count - k + 1)
			if i + k - 1 < s.count {
				
				end = s.index(s.startIndex, offsetBy: i + k - 1)
				print("in")
			}
			
			// print(i)
			var subString = String(s[start ... end])
			if subString.count < k {
				subString += repeatElement(fill, count: k - subString.count)
			}
			result.append(subString)
			
			i += k
		}
		
		return result
	}
}

File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.ts
function divideString(s: string, k: number, fill: string): string[] {
  let result: string[] = [];
  let str = "";
  let d = k - (s.length % k);
  if (d == k) {
    d = 0;
  }

  for (let i = 0; i < s.length + d; i++) {
    if (i < s.length) {
      const c = s[i];
      str += c;
    } else {
      str += fill;
    }
    if (str.length == k) {
      result.push(str);
      str = "";
    }
  }

  return result;
}
